service/src/main/java/com/ge/predix/acs/config/GraphConfig.java

```diff
@@ -135,38 +148,46 @@ public static void createUniqueIndexForLabel(final Graph newGraph, final String
         ManagementSystem.awaitGraphIndexStatus((TitanGraph) newGraph, indexName).status(SchemaStatus.ENABLED).call();
     }
 
-    public static void createTwoKeyUniqueCompositeIndex(final Graph newGraph, final String indexName,
-            final String indexKey1, final String indexKey2) throws InterruptedException {
+    public static void createUniqueCompositeIndex(final Graph newGraph, final String indexName,
+            final String[] propertyKeyNames) throws InterruptedException {
+        createUniqueCompositeIndex(newGraph, indexName, propertyKeyNames, String.class);
+    }
+
+    public static void createUniqueCompositeIndex(final Graph newGraph, final String indexName,
+            final String[] propertyKeyNames, final Class<?> propertyKeyType) throws InterruptedException {
```

Comment:
This method signature assumes that all property keys of the composite index are the same type. Will this always be the case? Does it make sense to provide property (keyName, keyType) tuples instead?